MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -1522,6 +1522,24 @@
                 setConfigurationResult(reportError(VK_ERROR_FEATURE_NOT_PRESENT, "vkCreateImageView(): 2D views on 3D images can only be used as color attachments."));
             }
         }
+
+        // If a 2D array view on a 2D image with layerCount 1, and the only usages are
+        // attachment usages, then force the use of a 2D non-arrayed view. This is important for
+        // input attachments, or they won't match the types declared in the fragment shader.
+        // Transfer usages are OK: the transfer commands don't use image views.
+        if (mvkIsOnlyAnyFlagEnabled(_usage, (VK_IMAGE_USAGE_TRANSFER_SRC_BIT |
+                                             VK_IMAGE_USAGE_TRANSFER_DST_BIT |
+                                             VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT |
+                                             VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT |
+                                             VK_IMAGE_USAGE_INPUT_ATTACHMENT_BIT |
+                                             VK_IMAGE_USAGE_TRANSIENT_ATTACHMENT_BIT))) {
```

Comment:
Does it matter whether the usage is for attachments only? If the Image is not-an-array, wouldn't it make sense to make the view also not-an-array for any usage?

Reply:
A shader could have an image variable (or a combined image-sampler) with an array type. This becomes a `texture2d_array<T>` in MSL. If we do this all the time, then the validation layer will complain about mismatched texture types when we bind a sampled or storage image view.

Reply:
I've added some comments explaining this.